### PR TITLE
Renamed swift::types to swift::file_types.

### PR DESCRIPTION
**Pull request by davidungar**
Per @jrose-apple suggestion, renamed swift::types to swift::file_types, because it is now in the Frontend, not just the Driver.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@swift-ci please smoke test linux platform

**Comment by davidungar** [comment]
@swift-ci please test compatibility

**Comment by davidungar** [comment]
@swift-ci please test source compatibility

File: include/swift/Frontend/FileTypes.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
+//===--- Types.h - Input & Temporary Driver Types ---------------*- C++ -*-===//
```
Comment (jrose-apple) [inline include/swift/Frontend/FileTypes.h]: Nitpick: the file header and include guards should be changed.

**Review by jrose-apple — APPROVED** [review]

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@swift-ci please smoke test linux platform

**Pull request merged**